str() fail on interface type

**V version:** V 0.1.28 288ea18
**OS:** macOS 10.15

**What did you do?**

Seems like the str method is not defined for the interface, how can I do that? ive seen the fn (g &Gen) interface_table() string { in cgen.v but i dont see where should i define this method to make the .str() code to work. And actually i dont know if this patch should be printing the name of the interface or the name of the type that is interfaced.

The problem is that right now i have 3 different compiler crashes because of that even without using the string representation of the interface. but i cant share it or make a small reproducer. will share it as soon as i get it

```go
$ v run turn.v
==================
^
/private/var/folders/fp/qqm5m5nj0ll0xtljv1306h5r0000gn/T/v/turn.tmp.c:7972:51: error: no member named 'str' in 'struct _Register_interface_methods'
        string b = Register_name_table[a._interface_idx].str(a._object);
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/private/var/folders/fp/qqm5m5nj0ll0xtljv1306h5r0000gn/T/v/turn.tmp.c:8010:38: warning: integer literal is too large to be represented in a signed integer type, interpreting as unsigned [-Wimplicitly-unsigned-literal]
        _const_math__bits__max_u64 = ((u64)(18446744073709551615));
                                            ^
/private/var/folders/fp/qqm5m5nj0ll0xtljv1306h5r0000gn/T/v/turn.tmp.c:8031:407: warning: integer literal is too large to be represented in a signed integer type, interpreting as unsigned [-Wimplicitly-unsigned-literal]
((u64)(1)), ((u64)(10)), ((u64)(100)), ((u64)(1000)), ((u64)(10000)), ((u64)(100000)), ((u64)(1000000)), ((u64)(10000000)), ((u64)(100000000)), ((u64)(1000000000)), ((u64)(10000000000)), ((u64)(100000000000)), ((u64)(1000000000000)), ((u64)(10000000000000)), ((u64)(100000000000000)), ((u64)(1000000000000000)), ((u64)(10000000000000000)), ((u64)(100000000000000000)), ((u64)(1000000000000000000)), ((u64)(10000000000000000000)),
                                                                                                                                                                                                                                                                                                                                                                                                                      ^
27 warnings and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

pmb:lala pancake$ cat turn.v
module main

interface Register {
        register()
}

struct Bar { }

fn (b Bar)register() { }

fn get_register() Register {
        return Bar{}
}

fn test_interface_string() {
        a := get_register()
	b := a.str()
        assert('$b' == 'Register')
}
fn main() { test_interface_string() }
```


**What did you expect to see?**

see above
 
**What did you see instead?**

see above

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -938,7 +938,7 @@ pub fn (mut c Checker) call_method(mut call_expr ast.CallExpr) table.Type {
 	if method_name == 'str' {
 		if left_type_sym.kind == .interface_ {
 			iname := left_type_sym.name
-			c.error('<$iname> interface does not have an .str() method. Use typeof() instead', call_expr.pos)
+			c.error('interface `$iname` does not have a .str() method. Use typeof() instead', call_expr.pos)
 		}
 		call_expr.receiver_type = left_type
 		call_expr.return_type = table.string_type

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -936,6 +936,10 @@ pub fn (mut c Checker) call_method(mut call_expr ast.CallExpr) table.Type {
 	}
 	// TODO: str methods
 	if method_name == 'str' {
+		if left_type_sym.kind == .interface_ {
+			iname := left_type_sym.name
+			c.error('<$iname> interface does not have an .str() method. Use typeof() instead', call_expr.pos)
+		}
 		call_expr.receiver_type = left_type
 		call_expr.return_type = table.string_type
 		if call_expr.args.len > 0 {

--- a/vlib/v/checker/tests/no_interface_str.out
+++ b/vlib/v/checker/tests/no_interface_str.out
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/no_interface_str.v:18:12: error: <Animal> interface does not have an .str() method. Use typeof() instead
+vlib/v/checker/tests/no_interface_str.v:18:12: error: interface `Animal` does not have a .str() method. Use typeof() instead
    16 | fn moin() {
    17 |     a := get_animal()
    18 |     println(a.str())

--- a/vlib/v/checker/tests/no_interface_str.out
+++ b/vlib/v/checker/tests/no_interface_str.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/no_interface_str.v:18:12: error: <Animal> interface does not have an .str() method. Use typeof() instead
+   16 | fn moin() {
+   17 |     a := get_animal()
+   18 |     println(a.str())
+      |               ~~~~~
+   19 | }

--- a/vlib/v/checker/tests/no_interface_str.vv
+++ b/vlib/v/checker/tests/no_interface_str.vv
@@ -1,0 +1,19 @@
+interface Animal {
+	speak()
+}
+
+struct Cow {
+}
+
+fn (c Cow)speak() {
+	println('moo')
+}
+
+fn get_animal() Animal {
+	return Cow{}
+}
+
+fn moin() {
+	a := get_animal()
+	println(a.str())
+}

